server/mergin/app.py

```diff
@@ -396,6 +396,15 @@ def log_bad_request(response):
 
         return response
 
+    @application.after_request
+    def set_custom_error_header(response):
+        """Clients (e.g. plugin) expect error `Content-Type` in response header.
+        Some responses with custom error lacks it."""
+        if response.status_code == 422:
```

Comment:
is this only for our custom errors or all aborts, like 404, 409, ...

Reply:
when the response is created with aborter, it has correct type

however, custom error responses are not always created with abort() as we want some custom attributes to add there
e.g. https://github.com/MerginMaps/server/blob/master/server/mergin/sync/public_api_controller.py#L662

We also return some custom errors in `207` but I'm not sure we want to change Content-Type headers for that.

Reply:
ah ok, thanks

207 is success-like, I would not worry about that one